### PR TITLE
Backport new pipelines to 2.3.x

### PR DESCRIPTION
**Pull request by elliedori**
This PR backports the new Github Actions PR & CI pipelines to 2.3.x

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -13,8 +13,8 @@ buildscript {
 	}
 
 	dependencies {
-		classpath 'io.spring.gradle:spring-build-conventions:0.0.28.RELEASE'
-		classpath "org.springframework.boot:spring-boot-gradle-plugin:$springBootVersion"
+		classpath 'io.spring.gradle:spring-build-conventions:0.0.34.RELEASE'
```
Comment (eleftherias) [inline build.gradle:16]: There are a couple more steps involved when updating from 0.0.28.RELEASE.
We've done the upgrade in master via 5acb307a54c7a5347ffce8b4be7adb9bd3339aa5, which you can use a reference.

Comment (eleftherias) [reply build.gradle:16]: I went ahead and backported the changes is 5acb307. If you rebase on master, the errors should be resolved.

**Comment by elliedori** [comment]
Spacing is fixed, ready for re-review @eleftherias

**Comment by eleftherias** [comment]
Closed via 1b63952e5841b3cae7c1db15ddc4aa4ed7433060 and fe32f7a66b3d2a8e52d7c7fb2453c67bf182dd26

**Pull request closed without merging**